### PR TITLE
Obj_Snowball2

### PR DESCRIPTION
**Pull request by Kelebek1**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

### CODE REVIEW

File: src/overlays/actors/ovl_Obj_Snowball2/z_obj_snowball2.c [hunk 1]
```diff
@@ -28,82 +38,589 @@ const ActorInit Obj_Snowball2_InitVars = {
...
+        spDC.x = temp_f22 * 4.0f;
+        spDC.y = (temp_f2 * 6.0f) + 9.0f;
+        spDC.z = temp_f24 * 4.0f;
+
+        spD0.x = (temp_f22 * temp_f20) + this->actor.world.pos.x;
+        spD0.y = (temp_f2 * temp_f20) + spC8;
+        spD0.z = (temp_f24 * temp_f20) + this->actor.world.pos.z;
+        temp = Rand_Next();
+
+        if ((i & 3) == 0) {
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Obj_Snowball2/z_obj_snowball2.c:174]: Would this match with `%4` instead? If not, could you change it to `!(i & 3)`?

File: src/overlays/actors/ovl_Obj_Snowball2/z_obj_snowball2.c [hunk 2]
```diff
@@ -28,82 +38,589 @@ const ActorInit Obj_Snowball2_InitVars = {
...
+            phi_v0 = 1;
+        } else {
+            phi_v0 = 0;
+        }
+
+        EffectSsKakera_Spawn(globalCtx, &spD0, &spDC, &spD0, -300, phi_s1, 30, 0, 0, (temp >> 0x1D) + 8, phi_v0, 0, 50,
+                             -1, 0xEF, D_80B3A91C[i & 3]);
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Obj_Snowball2/z_obj_snowball2.c:181]: Same question about `%4`

File: src/overlays/actors/ovl_Obj_Snowball2/z_obj_snowball2.c [hunk 3]
```diff
@@ -28,82 +38,589 @@ const ActorInit Obj_Snowball2_InitVars = {
...
+
+void func_80B39470(Actor* thisx, GlobalContext* globalCtx) {
+    ObjSnowball2* this = THIS;
+    Vec3f sp58;
+    s32 phi_s0;
+    s32 i;
+
+    sp58.y = this->actor.world.pos.y + this->actor.depthInWater;
+
+    for (phi_s0 = 0, i = 0; i < 5; i++, phi_s0 += 0x3333) {
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Obj_Snowball2/z_obj_snowball2.c:201]:
```suggestion
    for (phi_s0 = 0, i = 0; i < 5; i++, phi_s0 += 0x10000 / 5) {
```

File: src/overlays/actors/ovl_Obj_Snowball2/z_obj_snowball2.c (hunk 1 above)
Comment (Kelebek1) [reply src/overlays/actors/ovl_Obj_Snowball2/z_obj_snowball2.c:174]: Why `!`? it's not a bool.

File: src/overlays/actors/ovl_Obj_Snowball2/z_obj_snowball2.c [hunk 4]
```diff
@@ -28,82 +38,589 @@ const ActorInit Obj_Snowball2_InitVars = {
...
+    s32 sp38 = (this->collider.base.acFlags & AC_HIT) != 0;
+    s32 pad2;
+
+    if (sp38) {
+        this->collider.base.acFlags &= ~AC_HIT;
+    }
+
+    if (Actor_HasParent(&this->actor, globalCtx)) {
+        this->actor.room = -1;
+        this->actor.flags |= 0x10;
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Obj_Snowball2/z_obj_snowball2.c:355]:
```suggestion
        this->actor.flags |= ACTOR_FLAG_10;
```

File: src/overlays/actors/ovl_Obj_Snowball2/z_obj_snowball2.c [hunk 5]
```diff
@@ -28,82 +38,589 @@ const ActorInit Obj_Snowball2_InitVars = {
...
+}
+
+void ObjSnowball2_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    ObjSnowball2* this = THIS;
+
+    Collider_DestroyJntSph(globalCtx, &this->collider);
+}
+
+void func_80B39C78(ObjSnowball2* this) {
+    this->actor.flags |= 0x10;
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Obj_Snowball2/z_obj_snowball2.c:339]:
```suggestion
    this->actor.flags |= ACTOR_FLAG_10;
```

File: src/overlays/actors/ovl_Obj_Snowball2/z_obj_snowball2.c [hunk 6]
```diff
@@ -28,82 +38,589 @@ const ActorInit Obj_Snowball2_InitVars = {
...
+        func_80B3A498(this);
+    } else {
+        CollisionCheck_SetAT(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    }
+}
+
+void func_80B3A498(ObjSnowball2* this) {
+    this->actor.home.pos.x = this->actor.world.pos.x;
+    this->unk_1AC = 46;
+    this->actor.flags |= 0x10;
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Obj_Snowball2/z_obj_snowball2.c:527]:
```suggestion
    this->actor.flags |= ACTOR_FLAG_10;
```

File: src/overlays/actors/ovl_Obj_Snowball2/z_obj_snowball2.c [hunk 7]
```diff
@@ -28,82 +38,589 @@ const ActorInit Obj_Snowball2_InitVars = {
...
+    this->actor.home.pos.y = this->actor.world.pos.y + this->actor.depthInWater;
+    this->actor.home.pos.z = this->actor.world.pos.z;
+    this->actor.world.pos.y = this->actor.world.pos.y + (this->actor.shape.yOffset * this->actor.scale.y);
+    this->actor.shape.yOffset = 0.0f;
+    this->actor.speedXZ = 0.0f;
+    this->actionFunc = func_80B3A500;
+}
+
+void func_80B39638(GlobalContext* globalCtx, Vec3f* arg1);
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Obj_Snowball2/z_obj_snowball2.c:536]: This prototype is redundant, isn't it?

File: src/overlays/actors/ovl_Obj_Snowball2/z_obj_snowball2.c [hunk 8]
```diff
@@ -5,6 +5,7 @@
  */
 
 #include "z_obj_snowball2.h"
+#include "objects/object_goroiwa/object_goroiwa.h"
 
 #define FLAGS 0x00800000
```
Comment (hensldm) [inline src/overlays/actors/ovl_Obj_Snowball2/z_obj_snowball2.c:10]:
```suggestion
#define FLAGS (ACTOR_FLAG_800000)
```

File: src/overlays/actors/ovl_Obj_Snowball2/z_obj_snowball2.c [hunk 9]
```diff
@@ -28,82 +38,589 @@ const ActorInit Obj_Snowball2_InitVars = {
...
+            func_80B39908(this, globalCtx);
+        }
+
+        if (this->unk_1AD == 0) {
+            Actor_MoveWithGravity(&this->actor);
+            Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 15.0f, 15.0f, 0.0f, 0x44);
+            if ((this->actor.bgCheckFlags & 1) &&
+                (DynaPoly_GetActor(&globalCtx->colCtx, this->actor.floorBgId) == NULL)) {
+                this->unk_1AD = 1;
+                this->actor.flags &= ~0x10;
```
Comment (hensldm) [inline src/overlays/actors/ovl_Obj_Snowball2/z_obj_snowball2.c:382]:
```suggestion
                this->actor.flags &= ~ACTOR_FLAG_10;
```

**Review by hensldm — APPROVED** [review]
Couple minor things.

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**